components/menu/src/menu-item/menu-item.js

```diff
@@ -10,18 +10,26 @@ import styles from './menu-item.styles.js'
 const isModifiedEvent = (evt) =>
     evt.metaKey || evt.altKey || evt.ctrlKey || evt.shiftKey
 
-const createOnClickHandler =
-    ({ onClick, toggleSubMenu, isLink, value }) =>
-    (evt) => {
-        if ((isLink && isModifiedEvent(evt)) || !(onClick || toggleSubMenu)) {
-            return
-        }
-        evt.preventDefault()
-        evt.stopPropagation()
+const createOnClickHandler = ({
+    onClick,
+    openSubmenuOnHover,
+    toggleSubMenu,
+    isLink,
+    value,
+}) => (evt) => {
+    if ((isLink && isModifiedEvent(evt)) || !(onClick || toggleSubMenu)) {
+        return
+    }
+    evt.preventDefault()
+    evt.stopPropagation()
+
+    onClick && onClick({ value }, evt)
 
-        onClick && onClick({ value }, evt)
-        toggleSubMenu && toggleSubMenu()
+    if (!openSubmenuOnHover && toggleSubMenu) {
```

Comment:
Technically it doesn't because of this: https://github.com/dhis2/ui/blob/c48bf7d10766b4ce7a8c28aaadd752f3cfb47bb2/components/menu/src/menu-item/menu-item.js#L70-L80

Having said that, the way the code is structured now is confusing, I'll push a commit with  a cleaner implementation.